pheweb/serve/data_access/db.py

```diff
@@ -1450,5 +1472,15 @@ def get_UKBB_dao(self, singlematrix=False):
         else:
             return self.dao_impl["externalresult"]
 
+    def get_UKBB_availability(self)->Dict[str,bool]:
+        """
+        Returns:
+            (Dict[str,bool]): Dictionary with keys "externalresultmatrix" and "externalresult". Values are boolean values representing whether the UKBB DAOs are available.
+        """
+        retval = {}
+        retval["externalresultmatrix"] = "externalresultmatrix" in self.dao_impl
+        retval["externalresult"] = "externalresult" in self.dao_impl
+        return retval
```

Comment:
namedtuple would be better here so the fields are fixed.

Reply:
True. Another possibility would be to move the logic from [here](https://github.com/FINNGEN/pheweb/blob/6da6a611168c65d74218807dad32c151e3531012/pheweb/serve/server_jeeves.py#L30-L40) to the datafactory direct so jeeves would become something like
```python
class ServerJeeves(object):
    def __init__(self, server_conf):
        #...
        self.ukbb_dao, self.ukbb_matrixdao = self.dbs_fact.get_UKBB_objects()
        #...
```

and in factory have
```python
class DataFactory(object):
#...
    def get_UKBB_objects(self) -> Tuple[Optional[ExternalResultDb],Optional[ExternalResultDb]]:
        file_available = "externalresult" in self.dao_impl
        matrix_available = "externalresultmatrix" in self.dao_impl
        if file_available and matrix_available:
            return (self.dao_impl["externalresult"], self.dao_impl["externalresultmatrix"])
        elif file_available:
            return (self.dao_impl["externalresult"], self.dao_impl["externalresult"])
         elif matrix_available:
             return (self.dao_impl["externalresultmatrix"], self.dao_impl["externalresultmatrix"])
         else:
             return (None, None)
```

Then we don't have to create a new dictionary or namedtuple to tell which of those DAOs are available, and the decision of what ukbb dao to use would be on DataFactory. It would also be more difficult to accidentally use a wrong DAO.